### PR TITLE
Fix 31 conformance validation failures in IAM and STS

### PR DESCRIPTION
**Pull request by vieiralucas**
## Summary
- Add missing input validation for 20 IAM actions (ListAccessKeys, ListPolicies, ListRoles, ListUsers) covering UserName, Marker, MaxItems, PathPrefix, Scope, and PolicyUsageFilter constraints
- Add missing input validation for 11 STS actions (AssumeRole, AssumeRoleWithSAML, AssumeRoleWithWebIdentity) covering ExternalId, Policy, SourceIdentity, and ProviderId constraints
- Add 3 E2E tests verifying validation errors for MaxItems=0, empty PathPrefix, and 1-char ExternalId

## Test plan
- [x] `cargo build -p fakecloud-iam` passes
- [x] `cargo test -p fakecloud-iam` passes (21 unit tests)
- [x] `cargo clippy -p fakecloud-iam -- -D warnings` clean
- [x] `cargo test -p fakecloud-e2e --test iam -- --list` shows 27 tests (24 existing + 3 new)

---
## Summary by cubic
Add missing input validation for IAM List* and STS Assume* calls to match AWS. Fixes 31 conformance failures by returning ValidationException instead of HTTP 200 on bad inputs.

- **Bug Fixes**
  - IAM: `ListAccessKeys`, `ListPolicies`, `ListRoles`, `ListUsers` validate `UserName`, `Marker`, `MaxItems`, `PathPrefix`, `Scope`, `PolicyUsageFilter`.
  - STS: `AssumeRole`, `AssumeRoleWithSAML`, `AssumeRoleWithWebIdentity` validate `ExternalId`, `Policy`, `SourceIdentity`, `ProviderId`.
  - E2E: switch to SDK-based tests; assert ValidationException for `MaxItems=0`, empty `PathPrefix`, and 1-char `ExternalId`.

<sup>Written for commit f1dcdf80368b19918afd5abc1a60cf01c271f75a. Summary will update on new commits.</sup>

### CODE REVIEW

File: crates/fakecloud-e2e/tests/iam.rs [hunk 1]
```diff
@@ -839,6 +839,53 @@ async fn iam_tag_user() {
...
+#[tokio::test]
+async fn iam_list_policies_validates_max_items() {
+    let server = TestServer::start().await;
+    let output = server
+        .aws_cli(&["iam", "list-policies", "--max-items", "0"])
+        .await;
+    assert!(
+        !output.success(),
+        "MaxItems=0 should fail validation but got success"
+    );
```
[the remark this answers is not in the document]
Comment (vieiralucas) [reply crates/fakecloud-e2e/tests/iam.rs]: Fixed in 38b0f42 — tests now check for ValidationException specifically.

File: crates/fakecloud-iam/src/iam_service.rs [hunk 2]
```diff
@@ -879,6 +879,27 @@ impl IamService {
...
+            "marker",
+            req.query_params.get("Marker").map(|s| s.as_str()),
+            1,
+            320,
+        )?;
+        validate_optional_range_i64(
+            "maxItems",
+            req.query_params
+                .get("MaxItems")
+                .and_then(|v| v.parse::<i64>().ok()),
```
[the remark this answers is not in the document]
Comment (vieiralucas) [reply crates/fakecloud-iam/src/iam_service.rs:898]: Acknowledged — non-numeric parse bypass is a pre-existing pattern across all services. Will address in a follow-up with a dedicated parse-and-validate helper.
